File: .sling/libs/sushi/eqela/jk-net/0.35.0/jk.http.server.service.lua
jk = jk or {}

jk.http = jk.http or {}

jk.http.server = jk.http.server or {}

jk.http.server.service = jk.http.server.service or {}

jk.http.server.service.HTTPServerService = {}
jk.http.server.service.HTTPServerService.__index = jk.http.server.service.HTTPServerService
_vm:set_metatable(jk.http.server.service.HTTPServerService, {})

function jk.http.server.service.HTTPServerService._create()
	local v = _vm:set_metatable({}, jk.http.server.service.HTTPServerService)
	return v
end

function jk.http.server.service.HTTPServerService:_init()
	self._isClassInstance = true
	self._qualifiedClassName = self._qualifiedClassName or 'jk.http.server.service.HTTPServerService'
	self['_isType.jk.http.server.service.HTTPServerService'] = true
end

function jk.http.server.service.HTTPServerService:_construct0()
	jk.http.server.service.HTTPServerService._init(self)
	return self
end

function jk.http.server.service.HTTPServerService:execute(ctx, rootHandler)
	local ioManager = _g.jk.socket.io.IOManager:createDefault()
	if not (ioManager ~= nil) then
		do _g.jk.log.Log:error(ctx, "Failed to create IO manager") end
		do return false end
	end
	do
		local server = _g.jk.http.server.HTTPServer._construct0(_g.jk.http.server.HTTPServer._create())
		local port = _g.jk.env.EnvironmentVariable:get("PORT")
		if port ~= nil then
			do server:setPort(_g.jk.lang.String:toInteger(port)) end
		end
		if rootHandler ~= nil then
			do server:pushRequestHandler2(rootHandler) end
		end
		if not server:initialize1(ioManager, ctx) then
			do _g.jk.log.Log:error(ctx, "Failed to initialize HTTP server.") end
			do return false end
		end
		if not ioManager:execute(ctx) then
			do _g.jk.log.Log:error(ctx, "Failed to execute IO manager") end
			do return false end
		end
		do _g.jk.log.Log:info(ctx, "Successfully completed.") end
		do return true end
	end
end

File: .sling/libs/web5/eqela/jk-net/0.38.0/jk.http.server.service.js
let JkHttpServerServiceHTTPServerService = function() {};

JkHttpServerServiceHTTPServerService.prototype._t = { "JkHttpServerServiceHTTPServerService" : true };
JkHttpServerServiceHTTPServerService.prototype._tobj = JkHttpServerServiceHTTPServerService;

JkHttpServerServiceHTTPServerService.NEW = function() {
	var v = new JkHttpServerServiceHTTPServerService;
	return v.CTOR_JkHttpServerServiceHTTPServerService();
};

JkHttpServerServiceHTTPServerService.prototype.CTOR_JkHttpServerServiceHTTPServerService = function() {
	return this;
};

JkHttpServerServiceHTTPServerService.execute = function(ctx, rootHandler) {
	var ioManager = JkSocketIoIOManager.createDefault();
	if(!(ioManager != null)) {
		JkLogLog.error(ctx, "Failed to create IO manager");
		return false;
	}
	var server = JkHttpServerHTTPServer.NEW();
	var port = JkEnvEnvironmentVariable.get("PORT");
	if(port != null) {
		server.setPort((JkLangString.toInteger(port)));
	}
	if(rootHandler != null) {
		server.pushRequestHandler2(rootHandler);
	}
	if(!server.initialize1(ioManager, ctx)) {
		JkLogLog.error(ctx, "Failed to initialize HTTP server.");
		return false;
	}
	if(!ioManager.execute(ctx)) {
		JkLogLog.error(ctx, "Failed to execute IO manager");
		return false;
	}
	JkLogLog.info(ctx, "Successfully completed.");
	return true;
};

JkHttpServerServiceHTTPServerService.IS_INSTANCE = function(o) {
	return o != null && o._t != null && o._t["JkHttpServerServiceHTTPServerService"] === true;
};
